app/services/keycloak_oauth/authorizable_service.rb

```diff
@@ -0,0 +1,24 @@
+require 'net/http'
+
+module KeycloakOauth
+  class AuthorizableError < StandardError; end
+
+  class AuthorizableService
+    HTTP_SUCCESS_CODES = [Net::HTTPOK, Net::HTTPNoContent]
+    DEFAULT_CONTENT_TYPE = 'application/x-www-form-urlencoded'.freeze
+    AUTHORIZATION_HEADER = 'Authorization'.freeze
+
+    private
+
+    def parsed_response(http_response)
+      response = http_response.body.present? ? JSON.parse(http_response.body) : http_response.body
```

Comment:
What happens when the response is invalid (e.g. not valid JSON)?

Reply:
It would raise an error. I was thinking that in this case we would want the error to be raised, as it means that we probably need to update the gem to handle more responses. But I'm not sure, maybe we want to wrap the error under an `AuthorizableError` to make it more meaningful. What do you think?

Reply:
What I'm not sure yet is how the host application would be able to catch the error (e.g. showing the user a nice error message when something fails). Maybe wrapping under `AuthorizableError` would help catching the error.

Reply:
Well, I would say that it depends on what type of failure it is.

My idea is that if something completely crashes in our app, e.g. if the session does not have an access token, then we'd want this to be an actual 500, rather than hide it from the user. It means that something is very wrong and and we need to fix it.

If Keycloak replies with something other than 200, then I think we could display a message to the user. For example, I could see this working well when the user is trying to access something it does not have access to (and maybe keycloak replies with an authorization error). However, for the logout, I don't see how there could be a user error, since they are just clicking on the logout button (and the logout endpoint does not take any input from the user itself).

Maybe it would help me to understand your idea better if you gave me an example of a failure that we could present to the user?

(also, either way, I think we will need to come up with a larger plan regarding error handling at some point)

Reply:
I haven't specifically thought about any kind of error; I was just wondering if we want to catch errors here and provide the host application with more details about the error.

But since it's not specifically related to the logout, you're right - let's postpone this and discuss error handling later.